### DISCUSSION TITLE
Adding image to relational lists

### DISCUSSION THREAD

**Discussion by cspecter** [directus/directus#10708] [Archive]
Before we go down the path developing a custom extension I wanted to make sure it would be possible with the current extensions SDK. Our app is using Supabase for our database and for file storage. Supabase manages files via a private table that store the filename, folder and other metadata. We are referencing this table in our other tables to add files to our collections (either a single file as a many-to-one or multiple files via a join table as a many-to-many).

```
USER (profile_picture) -> FILES // many-to-one
PRODUCT (image_gallery) -> PRODUCT_GALLERY [join table] -> FILES // many-to-many
```

We have Directus set up and customized as our internal CMS and, for everything that is not a file, this is working great. Currently we only see the file name field in Directus (as it does not know that this string field is supposed to be an image) and thus we would like to, instead of displaying the name string, display the actual image file in the collection view, in the select-dropdown and the relational collection drawer.

Would this be possible using a Display extension for all use cases? Or would we need to make a custom Interface?

```javascript
// Files table
{
   id: string
   name: string // The actual file name including subfolder, i.e. "profile_pictures/abcde1234.jpg"
   folder: string // The root folder for the images.
}

/**
Supabase will append a url to the file name when using their API. 
For a public folder this might look like "https://<supabase account-id>.supabase.co/files/<folder-name>/". 
So we would need to append this to the name somehow.
*/
```

**Comment by rijkvanzanten** [comment]
I think you should be able to make a custom Supabase-Image display that effectively renders something like (pseudo):

```
<img :src="supabaseUrl" />

const supabaseUrl = `https://12345.supabase.co/files/${props.value.folder}/${props.value.file}`
```

there's no out of the box solution in Directus right now to have Supabase managed images displayed in app.

**Reply by cspecter** [reply]
OK, great. We will give it a try. Will report back if it works.
